ppr-ui/src/App.vue

```diff
@@ -362,46 +360,67 @@ export default class App extends Mixins(AuthMixin) {
   private resetFlags (): void {
     this.appReady = false
     this.haveData = false
-    this.errorDialog = false
+    this.errorDisplay = false
     this.payErrorDisplay = false
   }
 
   /** Fetches authorizations and verifies and stores roles. */
-  private async loadAuth (): Promise<any> {
+  private async loadAuth (): Promise<ErrorIF> {
     // save roles from the keycloak token
-    const authRoles = getKeycloakRoles()
-    if (authRoles && authRoles.length > 0) {
-      if (authRoles.includes('gov_account_user')) {
-        // if staff make call to check for sbc
-        const isSbc = await getSbcFromAuth()
-        this.setRoleSbc(isSbc)
-      }
-      if (!authRoles.includes('ppr')) {
-        throw new Error('No access to PPR')
+    let message = ''
+    let statusCode = StatusCodes.OK
+    try {
+      const authRoles = getKeycloakRoles()
+      if (authRoles && authRoles.length > 0) {
+        if (authRoles.includes('gov_account_user')) {
+          // if staff make call to check for sbc
+          const isSbc = await getSbcFromAuth()
+          this.setRoleSbc(isSbc)
+        }
+        if (!authRoles.includes('ppr')) {
+          throw new Error('No access to PPR')
+        }
+        this.setAuthRoles(authRoles)
+      } else {
+        throw new Error('Invalid auth roles')
       }
-      this.setAuthRoles(authRoles)
-    } else {
-      throw new Error('Invalid auth roles')
+    } catch (error) {
+      message = String(error)
+      statusCode = StatusCodes.UNAUTHORIZED
+    }
+    return {
+      category: ErrorCategories.ACCOUNT_ACCESS,
+      message: message,
+      statusCode: statusCode
     }
   }
 
   /** Fetches current user info and stores it. */
-  private async loadUserInfo (): Promise<any> {
+  private async loadUserInfo (): Promise<ErrorIF> {
     // auth api user info
     const response = await this.fetchCurrentUser()
+    let message = ''
+    let statusCode = response.status
     const userInfo: UserInfoIF = response?.data
-    if (userInfo) {
+    if (userInfo && statusCode === StatusCodes.OK) {
       // ppr api user settings
       const settings: UserSettingsIF = await getPPRUserSettings()
       userInfo.settings = settings
       this.setUserInfo(userInfo)
-      if (!settings || settings?.error) {
-        // error popup -> user may still continue
-        throw new Error('Invalid user settings')
+      if (settings?.error) {
+        message = 'Unable to get user settings.'
+        statusCode = settings.error.statusCode
       }
```

Comment:
Should we return out of here if no error?

Reply:
Nope -- if successful the obj returned be: { category:..., message: '', statusCode: 200 }